src/roiextractors/extractors/minian/miniansegmentationextractor.py

```diff
@@ -0,0 +1,360 @@
+        """
+        dataset = self._read_zarr_group("/A.zarr")
+        if dataset is None or "A" not in dataset:
+            raise ValueError("No image masks found in A.zarr dataset.")
+        else:
+            # Transpose to (height, width, weight) format because A.zarr is stored as (weight, height, width)
+            return np.transpose(dataset["A"], (1, 2, 0))
+
+    def _read_background_image_mask_from_zarr(self):
+        """Read the background image masks (b) from the zarr output.
+
+        b.zarr structure is expected to be like:
+        ├── b.zarr
+        │   ├── animal () <U7
+        │   ├── b (608, 608) float64
+        │   ├── height (608,) int64
+        │   ├── session () <U14
+        │   ├── session_id () <U9
+        │   ├── unit_id () int64
+        │   └── width (608,) in
+
+        Returns
+        -------
+        image_masks: numpy.ndarray
+            The image masks for each background components.
+        """
+        dataset = self._read_zarr_group("/b.zarr")
+        if dataset is None or "b" not in dataset:
+            raise ValueError("No background image masks found in b.zarr dataset.")
+        else:
+            return np.expand_dims(dataset["b"], axis=2)
+
+    def _read_trace_from_zarr_field(self, field: str):
+        """Read the traces specified by the field from the zarr object.
+
+        Parameters
+        ----------
+        field: str
+            The field to read from the zarr object.
+
+        Returns
+        -------
+        trace: numpy.ndarray
+            The traces specified by the field.
+        """
+        dataset = self._read_zarr_group(f"/{field}.zarr")
+
+        if dataset is None or field not in dataset:
+            return None
+        elif dataset[field].ndim == 2:
+            return np.transpose(dataset[field])
+        elif dataset[field].ndim == 1:
+            return np.expand_dims(dataset[field], axis=1)
+
+    def _read_timestamps_from_csv(self) -> np.ndarray:
+        """Extract timestamps corresponding to frame indexes (not downsampled) of the stored denoised trace.
+
+        Returns
+        -------
+        np.ndarray
+            The timestamps of the denoised trace.
+        """
+        if self._times is not None:
+            return self._times
+
+        csv_file = self.folder_path / "timeStamps.csv"
+        df = pd.read_csv(csv_file)
+        frame_indexes = self._read_zarr_group("/C.zarr/frame")
+        filtered_df = df[df["Frame Number"].isin(frame_indexes)]
+
+        return filtered_df["Time Stamp (ms)"].to_numpy() * 1e-3
+
+    def get_original_timestamps(self) -> np.ndarray:
+        """Get the original timestamps from the CSV file.
+
+        Returns
+        -------
+        np.ndarray
+            The original timestamps in seconds.
+        """
+
+        return self._read_timestamps_from_csv()
+
+    def sample_indices_to_time(self, sample_indices: Union[FloatType, np.ndarray]) -> Union[FloatType, np.ndarray]:
```

Comment:
```suggestion
    def sample_indices_to_time(self, sample_indices: FloatType |  np.ndarray) -> FloatType | np.ndarray:
```
I think that @pauladkisson prefers that